File: combination100.rb
def patterns(str = "1", i = 0, arr = (2..9).to_a)
  i >= arr.size and return str
  ['+', '-', ''].map { |op| patterns("#{str}#{op}#{arr[i]}", i + 1, arr) }.flatten
end

ans = patterns.select { |p| eval(p) == 100 }
puts ans

File: combination100.jl
function patterns(pat = "1", i = 1, arr = collect(2:9), ans = [])
    length(arr) < i && return push!(ans, pat)
    foreach(op -> patterns(pat * op * string(arr[i]), i + 1, arr, ans), ("+", "-", ""))
    return ans
end

@time ans = filter(x -> eval(Meta.parse(x)) == 100, patterns())
println(ans)
